cmd/kops-controller/pkg/server/server.go

```diff
@@ -175,6 +175,11 @@ func (s *Server) issueCert(name string, pubKey string, id *fi.VerifyResult, vali
 			CommonName:   fmt.Sprintf("system:node:%s", id.NodeName),
 			Organization: []string{rbac.NodesGroup},
 		}
+	case "kubelet-server":
+		issueReq.Subject = pkix.Name{
+			CommonName: id.NodeName,
+		}
```

Comment:
```suggestion
		}
		issueReq.AlternateNames = []string{id.NodeName}
```

Accepting hostnames in the CommonName is deprecated.